src/main/java/de/qabel/desktop/remoteFS/RemoteFSController.java

```diff
@@ -0,0 +1,179 @@
+import java.util.ResourceBundle;
+import java.util.UUID;
+
+
+public class RemoteFSController implements Initializable {
+
+    private ImageView folderIcon = new ImageView(
+            new Image(getClass().getResourceAsStream("/folder.png"))
+    );
+
+    private TreeItem<BoxObject> parentNode;
+    private List<BoxFolder> folders;
+
+    @FXML
+    private TreeTableView treeTable;
+    @FXML
+    private TreeTableColumn<BoxObject, String> nameColumn;
+    @FXML
+    private TreeTableColumn<BoxObject, String> sizeColumn;
+    @FXML
+    private TreeTableColumn<BoxObject, String> dateColumn;
+
+    @FXML
+    public void initialize(URL location, ResourceBundle resources) {
+
+
+        try {
+            BoxNavigation nav = createSetup();
+            uploadFilesAndFolders(nav);
+
+            TreeItem rootNodeWithChilds = calculateFolderStructure(nav);
+            treeTable.setRoot(rootNodeWithChilds);
+        } catch (QblStorageException e) {
+            e.printStackTrace();
+        }
+        calculateTableContent();
+
+
+    }
+
+    TreeItem calculateFolderStructure(BoxNavigation nav) throws QblStorageException {
+
+      //  BoxFolder rootFolder = new BoxFolder("block", "root Folder", new byte[16]);
+
+        TreeItem<BoxObject> rootNode = new TreeItem<>(new BoxFolder("block", "root Folder", new byte[16]));
+        parentNode = calculateSubFolderStructure(nav, rootNode, true);
+        parentNode.setExpanded(true);
+        return parentNode;
+    }
+
+    private TreeItem<BoxObject> calculateSubFolderStructure(
+            BoxNavigation nav,
+            TreeItem<BoxObject> TreeNode,
+            boolean first) throws QblStorageException {
+
+        BoxNavigation target;
+        if (first) {
+            target = nav;
+        } else {
+            target = nav.navigate((BoxFolder) TreeNode.getValue());
+        }
+
+        for (BoxFile file : target.listFiles()) {
+            TreeItem<BoxObject> BoxObjectTreeItem = new TreeItem<>((BoxObject) file);
+            TreeNode.getChildren().add(BoxObjectTreeItem);
+
+        }
+        for (BoxFolder subFolder : target.listFolders()) {
+
+            TreeItem<BoxObject> BoxObjectTreeItem = new TreeItem<>(subFolder);
+            TreeNode.getChildren().add(BoxObjectTreeItem);
+
+            calculateSubFolderStructure(nav, BoxObjectTreeItem, false);
+        }
+
+        return TreeNode;
+    }
+
+    private void calculateTableContent() {
+        nameColumn.setCellValueFactory(new BoxObjectCellValueFactory("name"));
+        sizeColumn.setCellValueFactory(new BoxObjectCellValueFactory("size"));
+        dateColumn.setCellValueFactory(new BoxObjectCellValueFactory("mtime"));
+
+        treeTable.getColumns().setAll(nameColumn, sizeColumn, dateColumn);
+    }
+
+    private void uploadFilesAndFolders(BoxNavigation nav) throws QblStorageException {
+        List<BoxFolder> allFolders = nav.listFolders();
+
+        for (BoxFolder fo : allFolders) {
+            nav.navigate(fo);
```

Comment:
Do you really have to navigate into a folder in order to delete it from the parent nav instance?